Makefile

```diff
@@ -1,162 +1,188 @@
+generate-type-doc:  ## Generate typedoc
+	@cd frontend && npm install && npm run doc
 
 .PHONY: generate-api-types
-generate-api-types:  ## Generate api-types.
-	cd frontend && npm run generate-api-types
+generate-api-types:  ## Generate api-types
+	@cd frontend && npm run generate-api-types
 
 .PHONY: psql-r
-psql-r:  ## Remote connect to postgis, uses $POSTGRES_USER and $POSTGRES_DB.
-	psql -h db -p 5432 -U $(POSTGRES_USER) $(POSTGRES_DB)
-
-.PHONY: pre-commit-a
-pre-commit-a:  ## Check all files with pre-commit.
-	pre-commit run --all-files
+psql-r:  ## Remote connect to postgis, uses $POSTGRES_USER and $POSTGRES_DB
+	@psql -h db -p 5432 -U $(POSTGRES_USER) $(POSTGRES_DB)
 
-.PHONY: distclean
-distclean: clean uninstall  ## Clean everything and uninstalls.
+.PHONY: pre-commit
+pre-commit:  ## Check all files with pre-commit
+	@pre-commit run --all-files
 
-.PHONY: clean
-clean: clean-frontend clean-backend clean-mdbook clean-storybook clean-scraper  ## Clean everything.
+.PHONY: docker-up
+docker-up:  ## Start a containerized dev environment
+	@cd .devcontainer && docker compose -p "permaplant_devcontainer" up
 
-.PHONY: clean-frontend
-clean-frontend:  ## Remove /frontend node_modules .
-	cd frontend && rm -rf node_modules
 
-.PHONY: clean-backend
-clean-backend:  ## Remove /backend/target .
-	cd backend && make clean
+# Install
 
-.PHONY: clean-mdbook
-clean-mdbook:  ## Remove /mdbook .
-	mdbook clean
-
-.PHONY: clean-scraper
-clean-scraper:  ## Remove /scraper/node_modules .
-	cd scraper && rm -rf node_modules
-
-.PHONY: clean-storybook
-clean-storybook:  ## Remove storybook static folder.
-	cd frontend && rm -rf storybook-static
 
 .PHONY: install
-install:  ## Install ALL dependencies within the source repo.
+install: install-backend install-frontend  ## Install ALL dependencies within the source repo
+	@echo "Installation completed."
+
+.PHONY: install-frontend
+install-frontend:  ## Install frontend
+	@echo "Checking if npm install is required..."
+	@if [ $(NODE_MODULES_NEWER) -eq 0 ]; then \
+		echo "Running 'npm install'..."; \
+		npm install --prefix frontend; \
+	else \
+		echo "No need to run 'npm install'. node_modules is up to date."; \
+	fi
+
+.PHONY: install-backend
+install-backend:  $(FAKE_FILE)  ## Install backend and mdbook deps
+	@echo "Checking if backend dependencies install is required..."
+
+$(FAKE_FILE):  Makefile
+	@echo $(FAKE_CONTENT) > $(FAKE_FILE)
 	cd backend && make install
 	cargo install mdbook mdbook-mermaid mdbook-linkcheck
 	cargo install --git https://github.com/ElektraInitiative/mdbook-generate-summary mdbook-generate-summary --locked
 
-.PHONY: uninstall
-uninstall:  ## Uninstall ALL dependencies within the source repo.
-	-cd backend && make uninstall
-	-cargo uninstall mdbook mdbook-mermaid mdbook-linkcheck
-	-cargo uninstall mdbook-generate-summary
-
 .PHONY: install-e2e
-install-e2e:
-	cd e2e && ./install.sh
+install-e2e:  ## Install e2e dependencies within this repo
```

Comment:
I would also include that in install (so that targets without specifier always do everything) if it does nothing if already done.